it/controllers/pensions/paymentsIntoPensions/PensionsTaxReliefNotClaimedControllerISpec.scala

```diff
@@ -16,257 +16,780 @@
-import views.PensionsTaxReliefNotClaimedTestSupport.Selectors
-
-
-class PensionsTaxReliefNotClaimedControllerISpec extends IntegrationTest with ViewHelpers with BeforeAndAfterEach with PensionsDatabaseHelper {
-
-  private def pensionsUsersData(pensionsCyaModel: PensionsCYAModel): PensionsUserData = {
-    PensionsUserDataBuilder.aPensionsUserData.copy(
-      isPriorSubmission = false,
-      pensions = pensionsCyaModel
-    )
-  }
-  override val userScenarios: Seq[UserScenario[_, _]] = Seq.empty
-
-  ".show" should {
-
-    "render the pensions where tax relief is not claimed question page with no pre-filled radio buttons if no CYA question data" which {
-      lazy val result: WSResponse = {
-        dropPensionsDB()
-        authoriseAgentOrIndividual(isAgent = false)
-        val pensionsViewModel = aPaymentsIntoPensionViewModel.copy(pensionTaxReliefNotClaimedQuestion = None)
-        insertCyaData(pensionsUsersData(aPensionsCYAModel.copy(paymentsIntoPension = pensionsViewModel)), aUserRequest)
-        urlGet(
-          fullUrl(
-            pensionTaxReliefNotClaimedUrl(taxYearEOY)), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
-      }
 
-      "has an OK status" in {
-        result.status shouldBe OK
+class PensionsTaxReliefNotClaimedControllerISpec
+  extends YesNoControllerSpec("/payments-into-pensions/no-tax-relief") {
+
+  private val selectorForFirstParagraph = "#main-content > div > div > p:nth-of-type(1)"
+  private val selectorForSecondParagraph = "#main-content > div > div > p:nth-of-type(2)"
+  private val selectorForSubHeading = "#main-content > div > div > form > div > fieldset > legend"
+
+  val minimalSessionDataToAccessThisPage: PensionsCYAModel = aPensionsCYAEmptyModel
+    .copy(paymentsIntoPension =
+      PaymentsIntoPensionViewModel(
+        rasPensionPaymentQuestion = Some(true),
+        totalRASPaymentsAndTaxRelief = Some(BigDecimal("1.00")),
+        oneOffRasPaymentPlusTaxReliefQuestion = Some(true),
+        totalOneOffRasPaymentPlusTaxRelief = Some(BigDecimal("1.00")),
+        totalPaymentsIntoRASQuestion = Some(true)
+      ))
+
+  "This page" when {
+    "requested to be shown" should {
+      "redirect to the expected page" when {
+        "the user has no stored session data at all" in {
+
+          val response = getPage(None)
+
+          response must haveStatus(SEE_OTHER)
+          response must haveALocationHeaderValue(PageRelativeURLs.paymentsIntoPensionsCYAPage)
+
+        }
+        "the user has a session with an empty pensions model" in {
+
+          val response = getPage(Some(pensionsUserData(aPensionsCYAEmptyModel)))
+
+          response must haveStatus(SEE_OTHER)
+          response must haveALocationHeaderValue(PageRelativeURLs.paymentsIntoPensionsReliefAtSourcePage)
+
+        }
+        "the user has a session with an insufficient pensions model" when {
+          "case one" in {
```

Comment:
I thnk we should expand on these case names for clarity

Reply:
OK. I'll make that change now.

Reply:
Changed.